vllm/v1/executor/multiproc_executor.py

```diff
@@ -0,0 +1,168 @@
+import atexit
+import time
+from typing import List, Optional, Tuple
+
+from vllm.config import VllmConfig
+from vllm.distributed.device_communicators.shm_broadcast import MessageQueue
+from vllm.executor.multiproc_worker_utils import (
+    set_multiprocessing_worker_envs)
+from vllm.logger import init_logger
+from vllm.utils import get_distributed_init_method, get_open_port
+from vllm.v1.executor.abstract import RPCParams
+from vllm.v1.outputs import ModelRunnerOutput
+from vllm.v1.worker.gpu_worker import WorkerProc, WorkerProcHandle
+
+logger = init_logger(__name__)
+
+
+class MultiprocExecutor:
+
+    def __init__(self, vllm_config: VllmConfig) -> None:
+        # Call self.shutdown at exit to clean up
+        # and ensure workers will be terminated.
+        atexit.register(self.shutdown)
+
+        self.vllm_config = vllm_config
+        self.parallel_config = vllm_config.parallel_config
+
+        self.world_size = self.parallel_config.world_size
+        tensor_parallel_size = self.parallel_config.tensor_parallel_size
+        assert self.world_size == tensor_parallel_size, (
+            f"world_size ({self.world_size}) must be equal to the "
+            f"tensor_parallel_size ({tensor_parallel_size}). "
+            f"Pipeline parallelism is not yet implemented in v1")
+
+        # Set multiprocessing envs that are common to V0 and V1
+        set_multiprocessing_worker_envs(self.parallel_config)
+
+        # Multiprocessing-based executor does not support multi-node setting.
+        # Since it only works for single node, we can use the loopback address
+        # 127.0.0.1 for communication.
+        distributed_init_method = get_distributed_init_method(
+            "127.0.0.1", get_open_port())
+
+        # Initialize worker and set up message queues for SchedulerOutputs
+        # and ModelRunnerOutputs
+        self.worker_request_mq = MessageQueue(self.world_size, self.world_size)
```

Comment:
nit: rename to `rpc_input_mq` or something like that? `request` is too specialized, especially when people integrate general `collective_rpc` functions.

Reply:
how about `rpc_broadcast_mq`? IMO `input` is tricky because we deal with both sides of the message queue, and it's helpful to be able to use one name consistently in the worker process and the engine core process